### PR TITLE
INTMDB-436: Update privatelink endpoint service resources timeout config

### PR DESCRIPTION
**Pull request by maastha**
## Description

Enable configurable timeout, and make consistent 2 hour defaulting timeout across dedicated, serverless, and privatelink_endpoint_service_data_federation_online_archive

Link to any related issue(s): https://jira.mongodb.org/browse/INTMDB-436 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [X] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go [hunk 1]
```diff
@@ -8,11 +8,12 @@ import (
 	"strings"
 	"time"
 
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:12]: you need to run go import

File: mongodbatlas/resource_mongodbatlas_privatelink_endpoint_service_data_federation_online_archive.go [hunk 2]
```diff
@@ -6,10 +6,12 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
+	"time"
+
+	matlas "go.mongodb.org/atlas/mongodbatlas"
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_privatelink_endpoint_service_data_federation_online_archive.go:12]: 👀

**Review by andreaangiolillo — COMMENTED** [review]
Let's wait to release 1.10.2 before merging this PR

File: mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go (hunk 1 above)
Comment (maastha) [reply mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:12]: @andreaangiolillo  I have that running, I guess that's why it reformatted. What are you expecting this to be? Maybe something wrong with my formatter settings

Comment (andreaangiolillo) [reply mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:12]: you need to remove the new line (line 12) and then run go import. Nothing wrong with your configuration, this happens all the time with go fmt

Comment (maastha) [reply mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:12]: This is how it gets formatted :/ , tried go fmt as well after removing all new lines/gaps. I guess it tries to group the imports

Comment (andreaangiolillo) [reply mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:12]: I don't know 😕  Thanks for the followup! I will resolve my comment

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go [hunk 3]
```diff
@@ -182,7 +183,7 @@ func resourceMongoDBAtlasPrivateLinkEndpointServerlessDelete(ctx context.Context
 		Refresh:    resourcePrivateLinkEndpointServerlessRefreshFunc(ctx, conn, projectID, instanceName, endpointID),
 		Timeout:    d.Timeout(schema.TimeoutDelete),
 		MinTimeout: 5 * time.Second,
-		Delay:      3 * time.Second,
+		Delay:      5 * time.Second,
```
Comment (Zuhairahmed) [inline mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:186]: just curious, why 5 seconds delay?

**Review by Zuhairahmed — APPROVED** [review]

File: mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go (hunk 3 above)
Comment (maastha) [reply mongodbatlas/resource_mongodbatlas_privatelink_endpoint_serverless.go:186]: @Zuhairahmed just making it consistent across privatelink_endpoint resources.

**Pull request merged**